src/MSBuild/XMake.cs

```diff
@@ -89,6 +89,13 @@ public enum ExitType
             ProjectCacheFailure
         }
 
+#if FEATURE_OSVERSION
+        /// <summary>
+        /// True if the Main method was invoked. False indicates that we're running hosted in another process (e.g. unit tests).
```

Comment:
Few things:
Can't you also check if Process.GetCurrentProcess().MainModule.FileName is msbuild.exe to know if we're hosted in something else?
We could be hosted in another process but spin out msbuild.exe worker nodes; wouldn't this mean they have the wrong idea?

Reply:
> Can't you also check if Process.GetCurrentProcess().MainModule.FileName is msbuild.exe to know if we're hosted in something else?

That would also work, with the small downside of potential false positives as anyone can name their process "MSBuild" but it is slower that setting/checking a flag. The host could be calling the `Main` method to circumvent the check but that's not something we support as a public API so I wouldn't worry about it much.

> We could be hosted in another process but spin out msbuild.exe worker nodes; wouldn't this mean they have the wrong idea?

MSBuild.exe has the correct manifest making Windows 10 (and later) not lie about its version, so if we're hosted and spin out an MSBuild.exe process, it's fine to be checking for `(10, 0)` there.